lib/iris/tests/__init__.py

```diff
@@ -527,6 +528,28 @@ def assertRaisesRegexp(self, *args, **kwargs):
         return six.assertRaisesRegex(super(IrisTest_nometa, self),
                                      *args, **kwargs)
 
+    @contextlib.contextmanager
+    def assertGivesWarning(self, expected_regexp='', expect_warning=True):
```

Comment:
This name could be confusing since this function can be used to check that code _doesn't_ give a warning.